src/PersonObjects/Player/PlayerObjectGeneralMethods.tsx

```diff
@@ -842,15 +838,13 @@ export function startFactionWork(this: IPlayer, router: IRouter, faction: Factio
   this.workRepGainRate *= BitNodeMultipliers.FactionWorkRepGain;
 
   this.isWorking = true;
-  this.focus = true;
   this.workType = CONSTANTS.WorkTypeFaction;
   this.currentWorkFactionName = faction.name;
 
   this.timeNeededToCompleteWork = CONSTANTS.MillisecondsPer20Hours;
-  router.toWork();
```

Comment:
Hmm, I could see this going wonky if the user is already focused / working say for a faction, and then a script changes them to working for a company.

I think in all these cases you need to check if the player is focused and still route to the corresponding work page if they were.

Reply:
The company and faction work pages are the same. When switching between the two like:
```js
/** @param {NS} ns **/
export async function main(ns) {
    ns.workForFaction("Tetrads", "security");
    ns.setFocus(true);
    await ns.sleep(5000);
    ns.workForCompany();
    await ns.sleep(10000);
    ns.stopAction();
}
```
In the background what happens is that the workForCompany function detects you were working and calls `singularityStopWork` which completes the current working operation, and then starts a new working operation. If you were on the Work page, the text updates automatically to be the Company work description.